packages/runtime-core/src/componentProxy.ts

```diff
@@ -42,28 +54,12 @@ export const PublicInstanceProxyHandlers = {
       // return the value from propsProxy for ref unwrapping and readonly
       return propsProxy![key]
     } else {
-      // TODO simplify this?
+      if (hasOwn(proxyHandlers, key)) {
+        return target[proxyHandlers[key]]
+      }
       switch (key) {
```

Comment:
This switch is no longer worth it and can be merged into the if/else branches as `else if (key === '$el') { ... } else { ... }`